lightningd/onchain_control.c

```diff
@@ -518,6 +518,43 @@ static u32 infinite_block_deadline(const struct chain_topology *topo)
 	return get_block_height(topo) + 300;
 }
 
+/**
+ * slow_sweep_dealine -- A smart(-ish) way of estimating sweep fees.
+ *
+ * We use the `close_blockheight` as the basis for the target. We then
+ * target to have the funds back at most 2 weeks after the close,
+ * linearly increasing the confirmation target until we get
+ * close. However, we also don't want to panic and set a target that
+ * is too close, in order not to waste too many funds on the sweep
+ * fees.
+ */
+static u32 slow_sweep_deadline(const struct chain_topology *topo, const struct channel *c)
+{
+	u32 deadline, height = get_block_height(topo);
+	u32 closeheight = *c->close_blockheight ? *c->close_blockheight : 0;
+	if (closeheight == 0)
+		return get_block_height(topo) + 300;
+
+	/* A two week deadline seems like a not too bad default. We
+	 * should likely tweak this to hit the desired cost vs. idle
+	 * time tradeoff. */
+	deadline = closeheight + 2016;
+
+	/* If we missed the deadline, don't panic, let's continue with
+	 * a somewhat reasonable target, and hope for the best. No
+	 * need to ramp up to confirmation target 1, we're supposed to
+	 * be slow. */
+	if (deadline < height + 12)
+		return height + 12;
+
+	/* At the same time don't be too lenient, we want this to
+	 * confirm and not just idle around forever. */
+	if (deadline - height > 300)
+		return height + 300;
```

Comment:
might be worth noting explicitly that the returned deadline value will always been within the range of [height + 12, height + 300] even though we're setting an eventual deadline of 2wks

Reply:
That's sort of mentioned in the function comment, isn't it? Well sans the 12 blocks upper limit on priority, but that's just for sanity.